homeassistant/components/nanoleaf/__init__.py

```diff
@@ -31,26 +35,40 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     nanoleaf = Nanoleaf(
         async_get_clientsession(hass), entry.data[CONF_HOST], entry.data[CONF_TOKEN]
     )
-    try:
-        await nanoleaf.get_info()
-    except Unavailable as err:
-        raise ConfigEntryNotReady from err
-    except InvalidToken as err:
-        raise ConfigEntryAuthFailed from err
-
-    async def _callback_update_light_state(event: StateEvent | EffectsEvent) -> None:
+
+    async def async_get_state() -> None:
+        """Get the state of the device."""
+        try:
+            await nanoleaf.get_info()
+        except Unavailable as err:
+            raise UpdateFailed from err
+        except InvalidToken as err:
+            raise ConfigEntryAuthFailed from err
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        logging.getLogger(__name__),
+        name=entry.title,
+        update_interval=timedelta(minutes=1),
+        update_method=async_get_state,
+    )
+
+    await coordinator.async_config_entry_first_refresh()
+
+    async def update_light_state_callback(event: StateEvent | EffectsEvent) -> None:
         """Receive state and effect event."""
-        async_dispatcher_send(hass, f"{DOMAIN}_update_light_{nanoleaf.serial_no}")
+        async_dispatcher_send(hass, f"{DOMAIN}_{nanoleaf.serial_no}_update_light")
```

Comment:
I think you can drop the dispatcher are as well since `async_set_updated_data` will callback the coordinator